src/storage.rs

```diff
@@ -161,9 +176,24 @@ where
     }
 
     // @TODO specify more useful error type
-    #[inline]
     fn init_active_blob(&mut self) -> Result<(), ()> {
-        self.active_blob = Some(Box::new(Default::default()));
+        let new_blob_name = format!(
+            "{}.{}.{}",
+            self.config.blob_file_name_prefix.as_ref().ok_or(())?,
+            self.blobs.len(),
```

Comment:
This is incorrect. Blob number should be calculated as the maximum from existing plus 1. If the old blob is deleted, the numbers will begin to conflict.

Reply:
Done